nexus/db-queries/src/db/datastore/address_lot.rs

```diff
@@ -210,50 +183,15 @@ impl DataStore {
     pub async fn address_lot_block_list(
         &self,
         opctx: &OpContext,
-        address_block_id: &NameOrId,
+        authz_address_lot: &authz::AddressLot,
         pagparams: &DataPageParams<'_, Uuid>,
     ) -> ListResultVec<AddressLotBlock> {
-        use db::schema::address_lot;
-        use db::schema::address_lot::dsl as lot_dsl;
-        use db::schema::address_lot_block::dsl as block_dsl;
+        use db::schema::address_lot_block::dsl;
 
         let pool = self.pool_authorized(opctx).await?;
 
-        let id = match &address_block_id {
-            NameOrId::Id(id) => *id,
-            NameOrId::Name(name) => {
-                let name = name.to_string();
```

Comment:
We don't want this decision happening at the datastore layer. In the http entrypoint we use the query and path parameters to retrieve a lookup object. This lookup is usually an authz resource that has a reference to the ID. Once we get to this layer we should already have the ID to work with. 

The caveat to the above is for list endpoints that do name or ID pagination.